test/integration/deploy_trigger_test.go

```diff
@@ -588,15 +589,19 @@ func TestTriggers_configChange(t *testing.T) {
 			t.Fatalf("expected watch event type %s, got %s", e, a)
 		}
 
-		if _, err := osClient.DeploymentConfigs(namespace).Update(config); err != nil {
-			return err
-		}
 		return nil
 	})
 	if retryErr != nil {
 		t.Fatal(retryErr)
 	}
 
+	if retryErr := kclient.RetryOnConflict(wait.Backoff{Steps: maxUpdateRetries}, func() error {
+		_, err := osClient.DeploymentConfigs(namespace).Update(newConfig)
```

Comment:
@kargakis the idea here is to retry faster (than going back to the controller) on conflict... so in case the retry controller shuffle the queue we don't have to wait longer than we need.